test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,229 @@
+*** Settings ***
+Documentation       Host IP and Certificate change test Suite
+
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-config.resource
+Resource            ../resources/microshift-process.resource
+
+Library             OperatingSystem
+Library             SSHLibrary
+Library             BuiltIn
+Library             String
+Library             random
+
+Suite Setup         Setup Suite
+Suite Teardown      Teardown Suite
+
+Test Tags           restart    slow
+
+*** Variables ***
+${HOSTS_FILE_PATH}  /etc/hosts
+${NETWORK_FILE_PATH}  /etc/NetworkManager/system-connections/ens3.nmconnection
+${EXTERNAL_SERVER_CERT}    /var/lib/microshift/certs/kube-apiserver-external-signer/kube-external-serving/server.crt
+
+
+*** Keywords ***
+Generate New IP
+    ${gateway}    ${rc1}=    Execute Command    ip route show | grep -i ens3 | grep -i default | awk '{print $3}'    return_stdout=True    return_rc=True
+    ${hostip}    ${rc2}=    Execute Command    ip route show | grep -i ens3 | grep -i default | awk '{print $9}'    return_stdout=True    return_rc=True
+
+    Should Be Equal As Integers    ${rc1}    0
+    Should Be Equal As Integers    ${rc2}    0
+
+    FOR    ${i}    IN RANGE    10    # Maximum 10 attempts to find an unassigned IP
+        ${last_digit}    Evaluate    int($hostip.split('.')[-1])
+        ${random_digit}    Evaluate    random.randint(0, 9)
+        ${new_host_ip}    Evaluate    '${hostip}'.rsplit('.', 1)[0] + '.23' + str($random_digit)
```

Comment:
Rather than picking a random digit, I would run through 1...254 and check which IP is available. At least, this way the procedure will be predictable.

Reply:
Also, this technique is somewhat problematic as we're running quite a few VMs in parallel. If the hypervisor assigns the same IP address to another VM, there's going to be a conflict and hard to debug failures.

Reply:
I would suggest changing the network device MAC address and rebooting the host to obtain a new IP

Reply:
Updated

Reply:
I think the code still requires a change. We cannot just change an IP from within the VM without hypervisor "knowing" about it.

Reply:
I tried  dhclient, on microshift vm we don't have dhclient command